<a href="https://colab.research.google.com/github/Abdullah200401/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Sagnik556/flyrank-ml-internship-starter/blob/main/work/notebooks/w01_research_question.ipynb)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I can use the available data to measure observed page-level signals and identify directional patterns that may help prioritize pages for review. The eventual output should be treated as decision-support for a content or SEO team. I cannot claim that a content refresh will cause higher traffic, CTR, or engagement based only on this dataset. I also cannot claim to predict Google's ranking decisions or guarantee that a recommended page will improve after being refreshed.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

I can use the available data to measure observed page-level signals and identify directional patterns that may help prioritize pages for review. The eventual output should be treated as decision-support for a content or SEO team. I cannot claim that a content refresh will cause higher traffic, CTR, or engagement based only on this dataset. I also cannot claim to predict Google's ranking decisions or guarantee that a recommended page will improve after being refreshed.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [7]:
import pandas as pd

url = "https://raw.githubusercontent.com/Abdullah200401/flyrank-ml-internship/main/data/raw/content_refresh_anonymized.csv"

df = pd.read_csv(url)

print("Number of rows:", len(df))
print("Number of columns:", len(df.columns))

display(df.head())
print("Average search volume:", df["search_volume"].mean())
print("Average CTR:", df["ctr"].mean())
print("Average engagement rate:", df["engagement_rate"].mean())

Number of rows: 30000
Number of columns: 44


,content_id,client_id,search_volume,competition,competition_level,cpc,content_type,main_intent,word_count,char_count,...,char_count_tier,ctr,avg_position,engagement_rate,scroll_rate,ai_traffic_pct,impression_tier,position_tier,trend_direction,trend_pct
0,content_304f48230142,client_f369cb89fc,10.0,0.67,HIGH,2.05,keyword article,transactional,3221.0,20457.0,...,15000-25000,0.76,10.6,5.88,4.55,0.0,good,striking,down,-41.4
1,content_a1fb4e703a9e,client_4e07408562,90.0,0.01,LOW,0.05,keyword article,informational,2481.0,15562.0,...,15000-25000,0.05,20.3,0.00,10.00,0.0,good,page_3_5,down,-57.7
2,content_9aa793d4d895,client_7f2253d7e2,0.0,0.00,LOW,0.00,keyword article,informational,3515.0,23643.0,...,15000-25000,0.09,36.5,0.00,28.57,0.0,good,page_3_5,down,-60.9
3,content_331d6c4de07b,client_19581e27de,10.0,0.00,LOW,0.00,keyword article,commercial,NaN,NaN,...,NaN,0.49,6.2,1.28,3.45,0.0,good,page_1,stable,-13.8
4,content_d99b7a2d90ca,client_3fdba35f04,0.0,0.00,LOW,0.00,keyword article,informational,2803.0,17469.0,...,15000-25000,0.13,44.0,0.00,24.29,0.0,good,page_3_5,down,-34.7


Average search volume: 158.88239139909922
Average CTR: 0.5107333333333334
Average engagement rate: 2.53452


The starter dataset contains 30,000 rows and 44 columns. The average search volume is 158.88, while the average CTR is 0.51 and the average engagement rate is 2.53. These numbers show that the dataset contains measurable search and engagement signals at the page level. This makes it reasonable to investigate whether these signals can help prioritize pages for possible content refresh, while recognizing that these averages alone do not prove that a refresh will improve performance.


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

I can use the available data to measure observed page-level signals and identify directional patterns that may help prioritize pages for review. The eventual output should be treated as decision-support for a content or SEO team. I cannot claim that a content refresh will cause higher traffic, CTR, or engagement based only on this dataset. I also cannot claim to predict Google's ranking decisions or guarantee that a recommended page will improve after being refreshed.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.